reactor-core/build.gradle

```diff
@@ -84,7 +74,7 @@ dependencies {
 	optional "org.slf4j:slf4j-api:$slf4jVersion"
 
 	//Optional Metrics
-	optional "io.micrometer:micrometer-core:$micrometerVersion"
+	compileOnly "io.micrometer:micrometer-core:$micrometerVersion"
```

Comment:
we should keep it as `optional`

Reply:
Can you elaborate? Does it have to do with how this is handled when published in maven repos?

Reply:
`optional` goes to the generated POM and signals that _at least_ `$micrometerVersion` should be used, while `compileOnly` would be removed from the POM and there is no hint which version we used

Reply:
Slight problem: the tests don't pass anymore :( 
Micrometer IS on the test classpath when using `optional` apparently...

Reply:
Also, have a look at https://repo1.maven.org/maven2/io/projectreactor/reactor-core/3.3.7.RELEASE/reactor-core-3.3.7.RELEASE.pom, no particular mention of micrometer

Reply:
Related question: why don't we put blockHound in the proper configuration created for the test-set? Would fix the compilation errors I'm seeing in IDEA as far as I understand.

Reply:
ok so after discussing `optional` and looking at the pom, we'll track and kinda document the fact that optional dependencies are not in the pom anymore in #2262. the `optional` scope from our local copy of `OptionalDependenciesPlugin` is simply not made part of the pom, whereas the `compileOnly` vanilla scope results in `<scope>provided</scope>` in the pom, which we don't want.

Reply:
>Related question: why don't we put blockHound in the proper configuration created for the test-set? Would fix the compilation errors I'm seeing in IDEA as far as I understand.

Created #2264

Reply:
Are you sure that `compileOnly` creates `<scope>provided</scope>`? It is not the same in other Gradle projects I've seen

Reply:
We just tested and indeed, it does not.